trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/IMemoryAdressing.java

```diff
@@ -0,0 +1,251 @@
+import de.uni_freiburg.informatik.ultimate.core.model.models.ILocation;
+import de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator.preferences.CACSLPreferenceInitializer.CheckMode;
+import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.Pair;
+import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.Triple;
+
+/**
+ * The interface defining the functions for the different addressing modes.
+ */
+public interface IMemoryAdressing {
+	/**
+	 * Constructs the required metadata for the selected addressing mode.
+	 *
+	 * @param requiredFeatures
+	 *            The Features that are currently needed for the program to be verified.
+	 * @return The metadata declarations.
+	 */
+	List<Declaration> constructMetaData(RequiredMemoryModelFeatures requiredFeatures);
+
+	/**
+	 * Returns a list of metadata declarations needed for the memory model infrastructure.
+	 *
+	 * @return The declarations.
+	 */
+	List<MemoryModelDeclarations> metaDataDeclarations();
+
+	/**
+	 * Constructs a list of expressions that are used in the specifications of malloc.
+	 *
+	 * @return The expressions.
+	 */
+	List<Pair<Expression, Set<VariableLHS>>> constructMallocSpecificationExpressions(ILocation tuLoc,
+			MemoryArea memoryArea, RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Constructs a list of expressions that are used in the specifications of dealloc.
+	 *
+	 * @return The expressions.
+	 */
+	List<Triple<Expression, Set<VariableLHS>, Boolean>> constructDeallocSpecificationExpressions(ILocation tuLoc,
+			RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Returns a list of statements that are part of Ultimate.Init.
+	 *
+	 * @return The statements.
+	 */
+	List<Statement> constructUltimateInitStatements(ILocation loc,
+			RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			MemoryModelDeclarationsHandler memoryModelDeclarationsHandler, BigInteger fixedAddressCounter);
+
+	/**
+	 * Constructs the expressions used in the specifications for allocInit.
+	 *
+	 * @return The expressions.
+	 */
+	List<Pair<Expression, Set<VariableLHS>>> constructAllocInitSpecificationExpressions(ILocation tuLoc,
+			RequiredMemoryModelFeatures requiredMemoryModelFeatures,
+			MemoryModelDeclarationsHandler memoryModelDeclarationsHandler);
+
+	/**
+	 * Add or subtracts a pointer and an integer.
+	 *
+	 * @return The calculated pointer.
+	 */
+	Expression doPointerArithmetic(final int operator, final ILocation loc, final Expression ptrAddress,
+			final RValue integer, final ICType valueType, final CPrimitive integerExpressionType);
+
+	/**
+	 * Returns the step size in which the base value of the initial allocations must be increased.
+	 *
+	 * @return The step size.
+	 */
+	BigInteger fixedAddressCounterCountingStep(final Expression size);
```

Comment:
Add prefix `get` to interface method name.

```suggestion
	BigInteger getFixedAddressCounterCountingStep(final Expression size);
```

Reply:
See [bf2a925](https://github.com/ultimate-pa/ultimate/commit/bf2a9251bf23c01929004e0a22a407d1c43a4f33)